### DISCUSSION TITLE
Is there a list of supported tag fields for MKV?

### DISCUSSION THREAD

**Discussion by Davey-101** [Xanashi/Icaros#125] [Q&A] [answered]
I want to add details of audio languages for each movie I have but it would help if I knew what fields were supported and passed to explorer.

Is there a list or am I misunderstanding how it works? I may end up using "comments", which wouldn't be a problem.

**Comment by Xanashi** [comment, marked as answer]
Take a look at my first and second comment in this thread, to see which MKV tags are currently supported in Icaros;
https://github.com/Xanashi/Icaros/issues/67

It should be mentioned that, in MKV it is possible to set the language of each track in the file, and Icaros will also show these
languages in the video, audio and subtitle track properties. e.g.

If you have a file with two AAC 2 channel tracks which is respectively English and Italian, it would show up as the following in Explorer:

**Audio tracks:** AAC 2.0 [Eng], AAC 2.0 [Ita]

**Reply by Davey-101** [reply]
That's really helpful thanks. Adding a language to each track in MKVToolNix would be a good solution but too time consuming. Maybe I'll write a command line script for it.

It'll be easier to pick one of the fields you list and use one of mp3tag's batch functions... or keep using filename hashtags like I do now.

Thanks again!

**Reply by Xanashi** [reply]
You're very welcome. 

A batch script using mkvpropedit could definitely accomplish this, but you could also check if any of the MKVToolNix third-party batch  tools may do the job:
https://mkvtoolnix.download/links.html

Adding the languages to the tracks themselves also allow most good players to auto-select the tracks with the preferred languages,
which can be really nice.
